examples/plotting/server/line_animate.py

```diff
@@ -24,6 +24,6 @@ def update(step):
 
 curdoc().add_periodic_callback(update, 50)
 
-session.show() # open the document in a browser
+session.show(p) # open the document in a browser
```

Comment:
I'm OK with requiring an argument to `show` I'm guessing it makes things much cleaner and simpler. But in that case lets print an informative message/exception if it is called with no arguments, that instructs about what is needed to be done now. 

Alternatively, we still don't have to keep `figure`/`Figure` to do something with "no args", we probably could just default to showing `curdoc().roots` (and still warning/deprecating)